### DISCUSSION TITLE
Octoprint Settings

### DISCUSSION THREAD

**Discussion by jjfs127** [Jyers/Marlin#46] [Q&A] [answered]
Do we still need to apply the changes in Octoprint based on the image: Octoprint_Settings.JPG found in the rewrite branch?

**Comment by t3chguy** [comment, marked as answer]
If you want the screen to change when the host (octoprint) is printing/paused/complete then yes

**Reply by DeNohvo** [reply]
Is this enabled for precomplied versions?

**Reply by Araforn** [reply]
I've tried it tonight. Firmware is really great but the precompiled version is not changing the printing screen for me.

**Reply by dafi87** [reply]
It's changes that have to go into your Octoprint config. They can't be in the (pre)compiled firmware.

**Reply by Araforn** [reply]
Ah OK thanks. I'm trying to find the Octoprint_Settings.JPG but no luck. Do you have a link maybe?. Thabks again.

**Reply by t3chguy** [reply]
https://github.com/Jyers/Marlin/blob/Ender_3_V2_Rewrite/Images/Octoprint%20Settings.JPG

**Reply by 00danny00** [reply]
Time remaining and percentage complete stay at 0 on display. (otherwise no issues) Added the code to octoprint, no change.

Is there something else I need to update?

**Reply by t3chguy** [reply]
And do you have anything sending M73 commands? PrusaSlicer injects them or you can use https://plugins.octoprint.org/plugins/m73progress/

**Reply by 00danny00** [reply]
Hmmm, added that plugin.  It's showing me info, but the percentage jumps around.  (ie- 20%, 120%, 180%, 25%, etc)

But it's a starting point, thanks for pointing me in a direction.

**Reply by jjfs127** [reply]
Had that as well and later pulled solved it for me. But I'm not sure if it was resolved or not.

**Reply by Schnabulation** [reply]
Was a bug in earlier releases - is fixed now, can confirm...

**Comment by sympotonic** [comment]
Hmm i want to see my Octoprint settings, what and where should i set it in Octoprint? I cant find the picture Octoprint_Settings.JPG as jjfs127 mention in his question. Do anyone have information of what we should put in Octoprint, to see info? progress, layer etc.

**Reply by t3chguy** [reply]
https://github.com/Jyers/Marlin/blob/Ender_3_V2_Rewrite/Images/Octoprint%20Settings.JPG

**Reply by sympotonic** [reply]
Oh that was fast. Thank you. What info does this gives me ?

**Reply by t3chguy** [reply]
It tells you what settings to set in OctoPrint so that it tells the printer what it is doing.

**Reply by jjfs127** [reply]
> Oh that was fast. Thank you. What info does this gives me ?

From Jyers in a previous reddit post: 

For the printer to recognize when octoprint start, stops or pauses a print, you have to add M75, M76, and M77 commands into octoprints GCODE scripts. There is a picture of these settings in the images folder on the GitHub page.

In order to get octoprint progress and remaining time updates you need to have "#define LCD_SET_PROGRESS_MANUALLY" in your configuration_adv.h and an octoprint plugin that sends M73 commands with the progress and time remaining. I recommend the "M73 Progress" plugin.

**Reply by sympotonic** [reply]
Yea =) i put it in and see. However i was curious on what info i could see on the screen from Octoprint. can i see printed layers like 55/700 or something like that =)

**Reply by Schnabulation** [reply]
Yes you can, but this needs another OctoPrint plugin: DisplayLayerProgress

**Reply by sympotonic** [reply]
I put this line in the buttom at the config.txt at the octoprint sd card/boot: define LCD_SET_PROGRESS_MANUALLY and installed the M73 Progress plugin. i hope thats correct?

**Reply by jjfs127** [reply]
`#define LCD_SET_PROGRESS_MANUALLY` is already defined in the Configuration_adv.h file, you don't need to add it, just uncomment if using the E2V2 configs https://github.com/Jyers/Marlin/blob/dd1113016554204be90a68a6da2b97bdcb823e12/Configuration%20Files/E3V2%20Templates/BL%20Touch%203x3%20High%20Speed/Configuration_adv.h#L1162

**Reply by sympotonic** [reply]
Ahh okay, well i dont use the configs, i just installed the compiled firmware. however it does now say layer indicator not found in file "xxxxxxx.gcode" - check layer patterin in display layer progress settings and reupload the file...

Should i add M73 to my starting gcode?

**Reply by sympotonic** [reply]
Ahh its properly because i added search and replace for octolapse with:

(;LAYER:).{1,}
G4 P1 so i have to change DisplayLayerProgress to G4 P1 under layer configuration however it dosnt look like same text in scripting? 1		[;\s*LAYER:\s*([0-9]+).*]		CURA

**Reply by sympotonic** [reply]
> Yes you can, but this needs another OctoPrint plugin: DisplayLayerProgress

Can you help here? I tried bunch of things, but i cant make it work. cause of octolapse i did:
search and replace in Cura 4.8
(;LAYER:).{1,}
G4 P

And in the DissplayLayerProgress i changed 1 [;\sLAYER:\s([0-9]+).*] CURA to 1 [;\sG4 P1:\s([0-9]+).*] CURA

but i still get same message: layer indicator not found in gcode - check layer pattering in display layer progress settings...

**Reply by Schnabulation** [reply]
So first of all: you need to define `LCD_SET_PROGRESS_MANUALLY` in Marlins configuration_adv.h file. I don‘t know if this is already set in the precompiled .bin - but let‘s asume it is.

Now, to take advantage of this you just install and enable the OctoPrint Plugin „M73 Progress“. This does send M73 GCODE to the printer so the percentage of the DWIN display is updated. That‘s all.

If you want a layer counter in OctoPrint you can use the plugin „DisplayLayerProgress“ but for this to work you need some info in the GCODE of your files. Normally Cura adds these automatically with `;LAYER:1` which also works by default with DisplayLayerProgress. If you changed this for OctoLapse (I don‘t know why you should - I also have OctoLapse installed and that works flawlessly), I believe you need to adjust the Regex for the layer indicator in the settings of DisplayLayerProgress.

**Comment by DeNohvo** [comment]
My octo print has the below code, is it safe to remove for just M77? When i cancel a print it still says host printing. The actual print job stops but the screen does not update,

`; disable motors
M84

;disable all heaters
{% snippet 'disable_hotends' %}
{% snippet 'disable_bed' %}
;disable fan
M106 S0

M77 ; End Print Screen`
